### PR TITLE
Fixes search for rules that have the default package

### PR DESCRIPTION
**Pull request by Rikkola**
Bug 1257819 - [GSS] (6.1.z) Rules are not populated in dropdown list in Test Scenarios in BRMS 6.1
Bug 1257817 - Rules are not populated in dropdown list in Test Scenarios in BRMS 6.1

This fix needs the following PRs:
https://github.com/droolsjbpm/kie-wb-common/pull/116
https://github.com/droolsjbpm/drools-wb/pull/78
https://github.com/droolsjbpm/jbpm-designer/pull/67
https://github.com/droolsjbpm/jbpm-form-modeler/pull/17

### CODE REVIEW

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/KieProjectServiceMock.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import org.guvnor.common.services.project.model.POM;
+import org.guvnor.common.services.project.model.Package;
+import org.guvnor.common.services.project.model.Project;
+import org.guvnor.common.services.workingset.client.model.WorkingSetSettings;
+import org.guvnor.structure.repositories.Repository;
+import org.kie.workbench.common.services.shared.project.KieProject;
+import org.kie.workbench.common.services.shared.project.KieProjectService;
+import org.uberfire.backend.vfs.Path;
+
+public class KieProjectServiceMock
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/KieProjectServiceMock.java]: Why the need for this Mock class?

Can't we just ```mock( KieProjectService.class )``` instead?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/main/java/org/kie/workbench/common/services/refactoring/backend/server/query/TermsCheck.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+package org.kie.workbench.common.services.refactoring.backend.server.query;
+
+import java.util.Set;
+
+import org.kie.workbench.common.services.refactoring.model.index.terms.IndexTerm;
+import org.kie.workbench.common.services.refactoring.model.index.terms.valueterms.ValueIndexTerm;
+
+class
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/main/java/org/kie/workbench/common/services/refactoring/backend/server/query/TermsCheck.java]: And what, pray tell, crazy code formatter splits lines like this!?!? ;)

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/RefactoringQueryServiceImplGeneralTest.java [hunk 3]
```diff
@@ -64,7 +64,7 @@ public void setupNamedQueriesProducer() {
     @Test
     public void testGetNamedQueries() throws IOException, InterruptedException {
         final RefactoringQueryService service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                 namedQueriesProducer );
+                                                                                 new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/RefactoringQueryServiceImplGeneralTest.java]: Did you format this code? It shows a mixture of spaces near brackets... one of which is inconsistent with our "regular" formatter... but not critical to this PR!

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findfieldtypes/FindFieldTypesQueryInvalidIndexTermsTest.java [hunk 4]
```diff
@@ -66,7 +67,7 @@ public void testFindFieldTypesQueryInvalidIndexTerms() throws IOException, Inter
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findfieldtypes/FindFieldTypesQueryInvalidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findfieldtypes/FindFieldTypesQueryValidIndexTermsTest.java [hunk 5]
```diff
@@ -67,7 +68,7 @@ public void testFindFieldTypesQueryValidIndexTerms() throws IOException, Interru
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findfieldtypes/FindFieldTypesQueryValidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findruleattributes/FindRuleAttributesQueryInvalidIndexTermsTest.java [hunk 6]
```diff
@@ -66,7 +67,7 @@ public void testQueryInvalidIndexTerms() throws IOException, InterruptedExceptio
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findruleattributes/FindRuleAttributesQueryInvalidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findruleattributes/FindRuleAttributesQueryValidIndexTermsTest.java [hunk 7]
```diff
@@ -67,7 +68,7 @@ public void testQueryValidIndexTerms() throws IOException, InterruptedException
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findruleattributes/FindRuleAttributesQueryValidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/FindRulesByProjectQueryInvalidIndexTermsTest.java [hunk 8]
```diff
@@ -66,7 +68,7 @@ public void testQueryInvalidIndexTerms() throws IOException, InterruptedExceptio
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/FindRulesByProjectQueryInvalidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findtypes/FindTypesQueryInvalidIndexTermsTest.java [hunk 9]
```diff
@@ -66,7 +67,7 @@ public void testFindTypesQueryInvalidIndexTerms() throws IOException, Interrupte
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findtypes/FindTypesQueryInvalidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findtypes/FindTypesQueryValidIndexTermsTest.java [hunk 10]
```diff
@@ -66,7 +67,7 @@ public void testFindTypesQueryValidIndexTerms() throws IOException, InterruptedE
         when( namedQueriesProducer.iterator() ).thenReturn( queries.iterator() );
 
         final RefactoringQueryServiceImpl service = new RefactoringQueryServiceImpl( getConfig(),
-                                                                                     namedQueriesProducer );
+                                                                                     new NamedQueries(namedQueriesProducer) );
```
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findtypes/FindTypesQueryValidIndexTermsTest.java]: Code formatting.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/KieProjectServiceMock.java (hunk 1 above)
Comment (Rikkola) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/findrules/KieProjectServiceMock.java]: Good point. Was following the habbit of making a mock class for the client side, but this is server side. I'll simplify this.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/RefactoringQueryServiceImplGeneralTest.java (hunk 3 above)
Comment (Rikkola) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/query/RefactoringQueryServiceImplGeneralTest.java]: I'll fix this and the other code format issues while changing the mock.

**Comment by Rikkola** [comment]
Going to do some clean up based on the comments @manstis did.

**Comment by Rikkola** [comment]
Fixes done.

**Comment by manstis** [comment]
+1 to merge (with the others)

**Comment by csadilek** [comment]
Thanks!

master: 1ea7ca2898632ae43957b2bdbf6a549502c0eb0f

We need the same series of PRs for 6.3.x, I guess ;(,

**Pull request closed without merging**